### ISSUE TITLE
[openhabcloud] sleepSocketIO(long delay) halts entire io.socket thread

### ISSUE THREAD

**Issue by lsafelix75** [openhab/openhab-addons#13209]
Hi 

This line of code in CloudClient.java causes a huge problem if another bundle is developed to use io.socket. Once this function is called in e.g openhabcloud add-on bundle due to connectivity issue, the delay function will affect io.socket  on another bundle. Please consider to use asynchronous way like runnable scheduler to prepare for delay. 

  ```
                  socket.close();
        -----> sleepSocketIO(delay);
                    socket.connect();
```

**Comment by ssalonen** [comment]
Discussion on this in community forums: https://community.openhab.org/t/solved-io-socket-2-difference-socket-instances-connect-to-2-different-cloud-servers-causing-internal-routing-confusion-within-oh3/137818/3?u=ssalonen

**Comment by lolodomo** [comment]
Closed by #13421
